packages/oid4vci-rest-client/src/types/IOID4VCIRestClient.ts

```diff
@@ -0,0 +1,23 @@
+import { IAgentContext, IPluginMethodMap, IResolver } from '@veramo/core'
+import { CredentialOfferFormat, Grant } from '@sphereon/oid4vci-common'
+
+export interface IOID4VCIRestClient extends IPluginMethodMap {
+  vciClientCreateOfferUri(args: IVCIClientCreateOfferUriRequestArgs, context: IRequiredContext): Promise<IVCIClientCreateOfferUriResponse>
+}
+
+export interface IVCIClientCreateOfferUriRequestArgs {
+  grants?: Grant
+  credentials?: (CredentialOfferFormat | string)[]
```

Comment:
any reason why credentials is optional?

Reply:
Bot credentials and grants should be mandatory

Reply:
both are mandatory now